toolkit/offchain/src/await_tx.rs

```diff
@@ -0,0 +1,192 @@
+use anyhow::anyhow;
+use cardano_serialization_lib::Transaction;
+use ogmios_client::query_ledger_state::QueryLedgerState;
+use sidechain_domain::McTxHash;
+use std::time::Duration;
+use tokio_retry::{strategy::FixedInterval, Retry};
+
+pub trait AwaitTx {
+	#[allow(async_fn_in_trait)]
+	async fn await_tx_output<Q: QueryLedgerState>(
+		&self,
+		query: &Q,
+		tx_id: McTxHash,
+		tx: &Transaction,
+	) -> anyhow::Result<()>;
+}
+
+pub struct FixedDelayRetries {
+	delay: Duration,
+	retries: usize,
+}
+
+impl FixedDelayRetries {
+	pub fn new(delay: Duration, retries: usize) -> Self {
+		Self { delay, retries }
+	}
+
+	pub fn two_minutes() -> Self {
+		Self { delay: Duration::from_secs(5), retries: 23 }
+	}
+}
+
+impl AwaitTx for FixedDelayRetries {
+	async fn await_tx_output<Q: QueryLedgerState>(
+		&self,
+		query: &Q,
+		tx_id: McTxHash,
+		tx: &Transaction,
+	) -> anyhow::Result<()> {
+		let strategy = FixedInterval::new(self.delay).take(self.retries);
+		let address = tx.body().outputs().get(0).address().to_bech32(None)?;
```

Comment:
We can also query for the first utxo of the transaction directly:
```
curl -H 'content-type: application/json' -X POST http://localhost:1337 -d \
'{"jsonrpc": "2.0",
"method":"queryLedgerState/utxo",
"params":{
    "outputReferences":[
        {"transaction": {"id": "f9a60946cf9ad9578c71260637bdd4cbdbde4ea6b2c0ae916809e1e0ce6a0b74"}, "index": 0}
     ]
}}'  | jq
```
and just test for non-emptiness.

Reply:
Oh well, this mean a lot for this PR, because without knowing tx outputs we can in fact move the check out of current methods and use at higher level:
```
fn high_level_op(){
  let tx_id = make_and_submit_tx()?;
  await_tx(tx_id)
}
```
Thank you very much!

Reply:
I've updated the PR with new ogmios trait for this query